ci: add Jenkinsfile and deploy script

jakubgs opened the pull request:
Necessary to build in our own CI and publish via Caddy-Git.

jakubgs committed:
add MIT license to repo and package.json

Signed-off-by: Jakub Sokołowski <jakub@status.im>

jakubgs committed:
ci: add Jenkinsfile and deploy script

Necessary to build in our own CI and publish via Caddy-Git.

Signed-off-by: Jakub Sokołowski <jakub@status.im>

oskarth commented on Jenkinsfile:
```diff
@@ -0,0 +1,47 @@
+    GIT_AUTHOR_NAME = 'status-im-auto'
+    GIT_AUTHOR_EMAIL = 'auto@status.im'
+    /* Dev site deployment. */
+    DEV_SITE = 'dev.vac.dev'
```
Huh? Status specs should be separate from Vac.dev

oskarth replied:
Typo maybe?

jakubgs replied:
Oh, that's just a leftover because I copied the `Jenkinsfile`. There is no `dev` in this case since this repo doesn't have a `develop` branch, I'll remove that.

jakubgs committed:
clean up docs headers, add layout to all

Splits docs to use three different `permalink` paths:
- `/raw/`
- `/draft/`
- `/stable/`

Otherwise docs like these clash and only one is accessible:
```
docs/draft/3-whisper-usage.md
docs/stable/3-whisper-usage.md
```
```
docs/draft/6-payloads.md
docs/stable/6-payloads.md
```

Signed-off-by: Jakub Sokołowski <jakub@status.im>

jakubgs commented:
I also split the docs to use three different `permalink` paths:
- `/raw/`
- `/draft/`
- `/stable/`

Otherwise docs like these clash and only one is accessible:
```
docs/draft/3-whisper-usage.md
docs/stable/3-whisper-usage.md
```
```
docs/draft/6-payloads.md
docs/stable/6-payloads.md
```

jakubgs committed:
add Creative Commons copyright to all docs

Signed-off-by: Jakub Sokołowski <jakub@status.im>

jakubgs commented:
This PR ties into this issue which references broken links: https://github.com/status-im/specs/issues/155

jakubgs commented:
I thought `/stable` was better because it's more clear about state, and `/spec` is redundant because the domain is already `specs.status.im`, but as Oskar points out that it changes URLs that might be embedded somewhere I've switched it back to `/spec` and renamed the folder from `stable` to `spec`.

jakubgs commented:
Also, moved `15/NOTIFICATIONS` to `draft` because it was never made stable.
